gcp_airflow_foundations/source_class/ftp_sftp_source.py

```diff
@@ -0,0 +1,255 @@
+from dataclasses import fields
+from os import X_OK
+from urllib.parse import urlparse
+import logging
+from dacite import from_dict
+from dataclasses import dataclass
+from datetime import datetime
+import shutil
+import os
+import glob
+
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.dummy import DummyOperator
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.sensors.python import PythonSensor
+
+from gcp_airflow_foundations.operators.api.hooks.sftp_hook import SFTPHook
+from gcp_airflow_foundations.operators.api.sensors.sftp_sensor import SFTPFilesExistenceSensor
+from gcp_airflow_foundations.source_class.ftp_source import GenericFileIngestionDagBuilder
+from gcp_airflow_foundations.common.gcp.load_builder import load_builder
+from gcp_airflow_foundations.base_class.sftp_table_config import SFTPTableConfig
+
+
+class SFTPFileIngestionDagBuilder(GenericFileIngestionDagBuilder):
```

Comment:
For methods, it is good to 
1) Make them private if they are not used/meant to be used by other classes 
2) Mark them as "override" if they extend methods from base class

Reply:
Will do... i always forget python even really uses the notion of private methods